src/engine/comp/player_controlled.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+typedef u32 EntityId;
```

Comment:
Use an include here rather than a 'fwd declaration' - the whole point of this typedef is we can change the data type of entity ID in 1 place & have the change affected everywhere (this goes for other components too)